Mono.Cecil.Cil/Symbols.cs

```
@@ -802,6 +809,45 @@ public ISymbolReader GetSymbolReader (ModuleDefinition module, string fileName)
 			return null;
 		}
 
+		public static SymbolReaderKind GetSymbolReaderKind(string fileName)
+		{
+			try
+			{
```

If we are going to do a try/catch all exceptions, would a Try method pattern be better to follow?  i.e. along the lines of Dictionary.TryGetValue.

So the signature would be `public static bool TryGetSymbolReaderKind(string fileName, out SymbolReaderKind  kind)`

Alternatively, maybe we could loosen up the catch?  What failure case are we trying to guard against?  No symbol files?  Not a managed assembly?

---

ReadAssembly can throw for various reasons (no file, corrupt file, not managed, etc).  I think having a "None" enum is simpler and easier to deal with than the Try method pattern for this case.  What value would the Try pattern have?  The caller would then need additional logic to detect the error and come to the same conclusion: no symbols.

---

The idea with the Try pattern is that it communicates that the method is never going to throw, even if you pass junk input.

I don't have a strong opinion on it.  I'm ok with leaving it as is.

---

I think either API approach works. I am partial to `Try` approach but we probably don't need to change this.